app/models/manager_refresh/target.rb

```diff
@@ -18,7 +18,17 @@ def initialize(association:, manager_ref:, manager: nil, manager_id: nil, event_
       @association = association
       @manager_ref = manager_ref
       @event_id    = event_id
-      @options     = options
+      @options     = options.except(:payload, :payload_id)
+
+      payload = options[:payload]
+      @options[:payload_id] = BinaryBlob.create!(:binary => payload).id if payload
+
+      payload_id = options[:payload_id]
+      if payload_id
+        payload_blob = BinaryBlob.find(payload_id)
+        @options[:payload] = payload_blob.binary
+        payload_blob.destroy
+      end
```

Comment:
maybe something like this would be clearer?
```ruby
if options[:payload]
  @options[:payload_id] = BinaryBlob.create!(:binary => options[:payload]).id
elsif options[:payload_id]
  payload_blob = BinaryBlob.find(options[:payload_id])
  @options[:payload] = payload_blob.binary
  payload_blob.destroy
end
```

Reply:
I would prefer the actual database access to be moved out of here.

But the `if` / `elsif` removes my concern that we are creating the `Blob` and then deleting it a few instructions later